admin/acceptance.py

```diff
@@ -79,11 +80,15 @@ def get_trial_environment(cluster):
     """
     return {
         'FLOCKER_ACCEPTANCE_CONTROL_NODE': cluster.control_node.address,
-        'FLOCKER_ACCEPTANCE_AGENT_NODES':
-            ':'.join(node.address for node in cluster.agent_nodes),
+        'FLOCKER_ACCEPTANCE_NUM_AGENT_NODES': str(len(cluster.agent_nodes)),
         'FLOCKER_ACCEPTANCE_VOLUME_BACKEND': cluster.dataset_backend.name,
         'FLOCKER_ACCEPTANCE_API_CERTIFICATES_PATH':
             cluster.certificates_path.path,
+        'FLOCKER_ACCEPTANCE_HOST_MAPPING': json.dumps({
+            node.private_address: node.address
+            for node in cluster.agent_nodes
+            if node.private_address is not None
```

Comment:
Ah, no, because the mapping isn't necessary given cluster API for case where there's no private/public mapping, OK.